primitives/core/src/offchain/storage.rs

```diff
@@ -58,3 +77,63 @@ impl OffchainStorage for InMemOffchainStorage {
 		}
 	}
 }
+
+
+
+
+/// Change to be applied to the offchain worker db in regards to a key.
+#[derive(Debug,Clone,Hash,Eq,PartialEq)]
+pub enum OffchainOverlayedChange {
+	/// Remove the data associated with the key
+	Remove,
+	/// Overwrite the value of an associated key
+	SetValue(Vec<u8>),
+}
+
+/// In-memory storage for offchain workers.
+#[derive(Debug, Clone)]
+pub struct OffchainOverlayedChanges {
```

Comment:
I think it would be best to implement this as an enum:
```rust
pub enum OffchainOverlayedChanges {
   Disabled,
   Enabled(HashMap<Vec<u8>>, OffchainOverlayedChange),
}
```
This way, in case we know the indexing API is defined all `set/remove` operations are simply no-ops instead of buffering all the data in memory, just to discard them later.